lib/ansible/module_utils/ec2.py

```diff
@@ -134,13 +135,15 @@ def _boto3_conn(conn_type=None, resource=None, region=None, endpoint=None, **par
                          'the conn_type parameter in the boto3_conn function '
                          'call')
 
-    if params.get('config'):
-        config = params.pop('config')
-        config.user_agent_extra = 'Ansible/{0}'.format(__version__)
-    else:
-        config = botocore.config.Config(
-            user_agent_extra='Ansible/{0}'.format(__version__),
-        )
+    config = botocore.config.Config(
+        user_agent_extra='Ansible/{0}'.format(__version__),
+    )
+
+    if params.get('config') is not None:
```

Comment:
Where's the config argument come from?

Reply:
boto3_conn takes **params, so modules have been able to set custom configurations by doing
```
boto_core_config = botocore.config.Config(retries={'mode': 'adaptive', 'max_attempts': 10})
connection = boto3_conn(module, conn_type='client', resource=resource, region=region, endpoint=ec2_url, config=boto_core_config, **aws_connect_params)
```
The old boto config file actually supported a decent amount of customization for different services, whereas botocore/boto3 do not in ~/.aws/config. It makes sense that a module would want to define a resilient default config for the particular service. I'm not sure off the top of my head if there are still things in our codebase using it (though there have been, and easily could be 3rd party things). I don't think it's a super common use case, but the prospect of breaking something in module_utils/ec2 gives me figurative hives.